### ISSUE TITLE
Stack overflow with lambda properties (fatal)

### ISSUE THREAD

**Issue by sshumaker** [kcl-lang/kcl#1078]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

Try compiling this fragment (I've minimized it to make it simpler).

```
schema RTS[_path: [str]]:
    path: [str] = _path

schema ListOfString[_path: [str]](RTS):
  path = _path
  at: (int)-> RTS = lambda index:int -> RTS {
    RTS(_path)
  }

schema Challenge[_path: [str]](RTS):
  path=_path
  clarifying_questions:ListOfString = ListOfString(_path) {}

challenge = Challenge(["challenge"])

schema lOP[_lhs: RTS]:
    path: [str] = _lhs.path


x = lOP(challenge.clarifying_questions)
```

### 2. What did you expect to see? (Required)
```
x:
  path:
  - challenge
```

### 3. What did you see instead (Required)
```
thread 'main' has overflowed its stack
fatal runtime error: stack overflow
```

### 4. What is your KCL components version? (Required)

kcl version 0.7.5

**Comment by Peefy** [comment]
Thank you for your feedback. I reproduce it for the KCL v0.8.0-alpha.1 version too with a simpler case

```
schema A:

schema B(A):
    f: () = lambda {}

schema C[A: A]:

x = C(B())
```

I will open a PR to fix today.

**Comment by Peefy** [comment]
Closed by #1079
